### PR TITLE
revert: Change directives out for components to set changeDetection

### PR DESCRIPTION
**Pull request by xelaint**
This reverts commit graycoreio/daffodil#208

Partially addresses #213

Reverting because of the error on having two components in one element (`proceed-to-checkout` and `add-cart`). Change detection is not an issue if component/directive has no @Inputs/@Outputs, so we are sticking to directives since they are supplements to the button component.

### CODE REVIEW

**Comment by lderrickable** [comment]
There should be an issue added to track the "refactor solution" though, right? I think we'll forget about it otherwise

**Comment by xelaint** [comment]
@lderrickable created #213

**Comment by xelaint** [comment]
@damienwebdev @lderrickable squashed w revert commit message. ready for review.

File: apps/demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.spec.ts [hunk 1]
```diff
@@ -139,8 +133,8 @@ describe('CartWrapper', () => {
       cartComponent = fixture.debugElement.query(By.css('demo-cart'));
       cartTotalsComponent = fixture.debugElement.query(By.css('demo-cart-totals'));
       helpBoxComponent = fixture.debugElement.query(By.css('demo-help-box'));
-      proceedToCheckoutComponent = fixture.debugElement.query(By.css('[demo-proceed-to-checkout]'));
-      continueShoppingComponent = fixture.debugElement.query(By.css('[demo-continue-shopping]'));
+      proceedToCheckoutComponent = fixture.debugElement.query(By.css('[demoProceedToCheckout]'));
+      continueShoppingComponent = fixture.debugElement.query(By.css('[demoContinueShopping]'));
```
Comment (lderrickable) [inline apps/demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.spec.ts]: should be continueShoppingDirective and proceedToCheckoutDirective

Comment (xelaint) [reply apps/demo/src/app/cart/components/cart-wrapper/cart-wrapper.component.spec.ts]: resolved in 0125e29

**Comment by lderrickable** [comment]
lgtm

**Comment by xelaint** [comment]
@lderrickable you didn't approve the review

**Pull request merged**